analysis/spectra_chisq/_model_spectra.py

```diff
@@ -0,0 +1,475 @@
+"""The ``_model_spectra`` module models spectra by varying the dust extinction from the Milky Way.
+
+The model used is ``sncosmo.CCM89Dust``. The parameters A_v and amplitude are fit to the observed spectra. The best fit
+parameters are determined by ``scipy.optimize.curve_fit``
+The fit parameters are saved to a fits file.
+
+Currently, run this module at the command line using: "python _model_spectra.py"
+"""
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+import sncosmo
+from astropy.table import Table
+from functools import partial
+from scipy.optimize import curve_fit
+from sndata.csp import DR1, DR3
+
+from astropy.table import join, vstack #unique
+
+#from ._synthetic_photometry import get_synth_photo_for_id # FIXME uncomment if fit performed using photometry
+
+dr1, dr3 = DR1(), DR3()
+dr1.download_module_data()
+dr3.download_module_data()
+dr3.register_filters(force=True)
+
+
+def mjd_to_jd(time):
+    """Convert MJD time to JD
+    JD = MJD + 2400000.5
+    
+    Args:
+        time (float): JD time
+        
+    Returns:
+        (float) MJD time
+    """
+    
+    return time + 2400000.0
+
+
+def get_jd_t0(obj_id, tbl_num=3, mjd=True):
+    """Get time of peak brightness of SN in B-band
+    
+    Args:
+        obj_id (str): specific SN object id
+        tbl_num (int): `3` refers to Table 3 in DR3 paper (The Carnegie Supernova Project I:
+            Third Photometry Data Release of Low-Redshift Type Ia Supernovae and Other White Dwarf Explosions)
+        mjd (bool): True if the time given by `tbl_num` is MJD. False if it is JD
+    
+    Returns:
+        The JD time of max brightness as a float
+    """
+
+    tbl = dr3.load_table(tbl_num)
+    
+    # Get data for specific SN
+    obj_dat = tbl[tbl['SN'] == obj_id]
+    # Get t0 for specific SN
+    t0 = obj_dat['T(Bmax)'] # MJD, see DR3
+    if mjd:
+        # Convert t0 to JD time
+        return mjd_to_jd(time=t0)
+    if not mjd:
+        return t0
+
+
+def create_model(model, obj_id, a_v, redshift, amplitude, r_v=3.1):
```

Comment:
The name of this function is misleading and the docstring is incorrect. Generally speaking, this function is a small wrapper for `model.set` and is unnecessary. Eliminate this function.